hack/gatewayapi-conformance.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# Get Gateway API CRD bundle-version
+BUNDLE_VERSION=$(oc get crds gateways.gateway.networking.k8s.io -ojson | jq -r '.metadata.annotations."gateway.networking.k8s.io/bundle-version"')
+
+if [[ "${BUNDLE_VERSION}" == "null" ]]; then
+    echo "Cannot find bundle-version annotation from Gateway API CRD"
+    exit 1
+fi
+echo "Gateway API CRD bundle-version: ${BUNDLE_VERSION}"
+
+echo "Create GatewayClass gateway-conformance"
+oc apply -f -<<EOF
+apiVersion: gateway.networking.k8s.io/v1
+kind: GatewayClass
+metadata:
+  name: gateway-conformance
+spec:
+  controllerName: openshift.io/gateway-controller
+EOF
+
+oc wait --for=condition=Accepted=true gatewayclass/gateway-conformance --timeout=300s
+
+echo "All gatewayclass status:"
+oc get gatewayclass -A
+
+echo "Go version: $(go version)"
+cd ../..
+mkdir -p kubernetes-sigs
+cd kubernetes-sigs
+rm -rf gateway-api
```

Comment:
In order not to impact the working directory tree of the script runner - it's better to create a temporary directory using `mktemp` and clone gateway-api repository into it.
```suggestion
CLONE_DIR=$(mktemp -d)
cd "${CLONE_DIR}"
```